mdagent/tools/base_tools/analysis_tools/pca_tools.py

```diff
@@ -0,0 +1,370 @@
+from typing import Optional
+
+import matplotlib.pyplot as plt
+import mdtraj as md
+import numpy as np
+import pandas as pd
+import scipy
+import seaborn as sns
+from langchain.tools import BaseTool
+from pydantic import BaseModel, Field
+from sklearn.decomposition import PCA
+
+from mdagent.utils import FileType, PathRegistry
+
+
+class PCA_analysis:
+    def __init__(
+        self,
+        path_registry,
+        pc_percentage,
+        top_path,
+        traj_path,
+        sim_id,
+        selection="backbone",
+    ):
+        self.path_registry = path_registry
+        self.traj = md.load(traj_path, top=top_path)
+        self.sim_id = sim_id
+        if pc_percentage > 1:
+            pc_percentage /= 100
+        self.pc_percentage = pc_percentage
+        self.atom_indices = self.traj.top.select(selection)
+        self.selection = selection
+        self.pc = PCA(n_components=30)
+        self.reduce_dim = None
+        self._sub_pcs = None
+        self.n_pcs = None
+
+    def _align_trajectory(self):
+        ###########MD TRAJ ALIGNMENT################
+        try:
+            self.traj.superpose(self.traj, frame=0, atom_indices=self.atom_indices)
+            return "Trajectory aligned to the first frame. "
+        except Exception as e:
+            print(f"Error aligning trajectory: {str(e)}")
+
+    def get_pc(self):
+        align_message = self._align_trajectory()
+        # self.pc = pca.PCA(
```

Comment:
can we get rid of the commented out pieces or do we need them for some reason?

Reply:
thanks! i removed a ton from the MDanalysis functions, but I guess some went unnoticed be me.